var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -788,13 +828,22 @@ def configure(self, spec, prefix):
             # Not currently working for qt@5
             config_args.extend(["-device-option", "QMAKE_APPLE_DEVICE_ARCHS=arm64"])
 
+        if IS_WINDOWS:
+            global configure
```

Comment:
(question) I think you're setting this because `build_environment.set_package_py_globals` sets `__module__.configure`:

* I would have thought this would shadow it, so that `global configure` would not be necessary
* we should probably skip that logic in `build_environment` on Windows (or do you know of packages that make use of `./configure`?)

Reply:
If you set a global variable in a local scope hidden behind some conditional, python eclipses the global variable with the local one of the same name, but if the logic isn't hit (i.e. because you're not on Windows), you are then referencing an unbounded local variable, which is an error.

> we should probably skip that logic in build_environment on Windows (or do you know of packages that make use of ./configure?)

There are packages that use configure, they're just not autotools packages, and I think each time I define an executable for them anyway. We definitely could stop this from being defined on non Windows, but that adds more logic to reason about elsewhere and all it achieves here is removing the `global` statement.

Reply:
> all it achieves here is removing the global statement.

FWIW I *do* think that's a win: having to address these sorts of considerations in the package definition is an example of `build_environment` getting in the way (so IMO it's worth the extra consideration in `build_environment`, e.g. by only setting `__module__.configure` if the source dir has a `./configure` file). However, that doesn't have to be the focus of this PR (which is dealing with established behavior).

Reply:
> by only setting __module__.configure if the source dir has a ./configure file

`build_envrironment` is only run prior to staging, so are we able to reason about this at that time beyond a check for an autotools package type (from a grep, not every package that uses `m.configure` is an autotools package).
For now I'm going to tuck this behind an "if Windows" and create a follow-up PR to handle the alternative approach w.r.t checking for an autotools/configure build.